internal/spoke_k8s_client/spoke_k8s_client_manager.go

```diff
@@ -0,0 +1,44 @@
+package spoke_k8s_client
+
+import (
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+)
+
+// SpokeK8sClientManager maintains a set of spoke clients
+
+//go:generate mockgen --build_flags=--mod=mod -package=spoke_k8s_client -destination=mock_spoke_k8s_client_manager.go . SpokeK8sClientManager
+type SpokeK8sClientManager interface {
+	AddFromSecret(secret *corev1.Secret) (SpokeK8sClient, error)
+}
+
+type spokeK8sClientManager struct {
+	clientFactory SpokeK8sClientFactory
+	clientMap     map[string]SpokeK8sClient
+}
+
+func NewSpokeK8sClientManager(log logrus.FieldLogger) SpokeK8sClientManager {
+	return &spokeK8sClientManager{
+		clientFactory: NewSpokeK8sClientFactory(log),
+		clientMap:     make(map[string]SpokeK8sClient),
+	}
+}
+
+// Add a SpokeK8sClient to the client map (create the client using specified secret).
+// If already created, return the existing client.
```

Comment:
This describes to different behaviors. It adds a client, but also returns a client regardless of whether it was added just now or not. WDYT about renaming this `FromSecret`, so it does not make implications about when or how the client was created?